convert/convert_windows.ml

```diff
@@ -396,9 +396,59 @@ let convert (g : G.guestfs) source inspect i_firmware
   and configure_pnputil_install () =
     let fb_script = "@echo off\n\
                      \n\
-                     echo Wait for VirtIO drivers to be installed\n\
-                     %systemroot%\\Sysnative\\PnPutil -i -a \
-                     %systemroot%\\Drivers\\Virtio\\*.inf" in
+                     setlocal EnableDelayedExpansion\n\
+                     set log=%~dp0pnputil.log\n\
+                     set inf_dir=%systemroot%\\Drivers\\Virtio\\\n\
+                     echo Installing drivers from %inf_dir% > \"%log%\"\n\
+                     set FAILED=0\n\
+                     set REBOOT_PENDING=0\n\
+                     \n\
+                     timeout /t 10 /nobreak >nul\n\
+                     \n\
+                     reg query \"HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\WindowsUpdate\\Auto Update\\RebootRequired\" >nul 2>&1\n\
+                     if %errorlevel%==0 (\n\
+                     echo Windows Update: Reboot required.\n\
+                     set REBOOT_PENDING=1\n\
+                     )\n\
+                     \n\
+                     reg query \"HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Component Based Servicing\\RebootPending\" >nul 2>&1\n\
+                     if %errorlevel%==0 (\n\
+                     echo CBS: Reboot required.\n\
+                     set REBOOT_PENDING=1\n\
+                     )\n\
+                     \n\
+                     reg query \"HKLM\\SYSTEM\\CurrentControlSet\\Control\\Session Manager\" /v PendingFileRenameOperations >nul 2>&1\n\
+                     if %errorlevel%==0 (\n\
+                     echo Session Manager: Reboot required.\n\
+                     set REBOOT_PENDING=1\n\
+                     )\n\
+                     \n\
+                     if \"%REBOOT_PENDING%\"==\"1\" (\n\
+                     echo A reboot is pending.\n\
+                     exit /b 1\n\
```

Comment:
The problem here is that using a common error return like `1` will cause other scripts that fail to also be retried.  Current behaviour is to log this and continue with the next script.  However those scripts will now cause reboot loops which isn't desirable.

Solution would be to use a special code (eg. `99`, I've no idea what range Windows permits). The special code would mean "retry this script again".